nsdb-common/src/main/scala/io/radicalbit/nsdb/common/statement/SQLStatement.scala

```diff
@@ -32,24 +32,19 @@ final case class Field(name: String, aggregation: Option[Aggregation])
   * [[ListFields]] if a list of fields are specified in the query.
   */
 sealed trait SelectedFields
-case object AllFields                            extends SelectedFields
+case class AllFields()                           extends SelectedFields
 final case class ListFields(fields: List[Field]) extends SelectedFields
 
 final case class ListAssignment(fields: Map[String, JSerializable])
 final case class Condition(expression: Expression)
 
-/**
-  * Where condition expression for queries.
-  * [[ComparisonExpression]] simple comparison expression.
-  */
 sealed trait Expression
 
 /**
-  * Simple expression having a [[SingleLogicalOperator]] applied.
+  * Simple expression having a [[NotOperator]] applied.
   * @param expression the simple expression.
-  * @param operator the operator to be applied to the expression.
   */
-final case class UnaryLogicalExpression(expression: Expression, operator: SingleLogicalOperator) extends Expression
+final case class UnaryLogicalExpression(expression: Expression) extends Expression
```

Comment:
I think that for clarity it's better to rename `UnaryLogicalExpression` into `NotExpression` or something like that.

Reply:
Agreed